docs(date-range-picker): add date range picker topic (#1525)
IgniteUI/igniteui-xplat-docs: doc/en/components/scheduling/date-range-picker.md

### Getting Started

<!-- WebComponents -->

To start using the `DateRangePicker`, you first need to install the Ignite UI for Web Components by running the following command:

```cmd
npm install {PackageWebComponents}
```

After that, you need to import the `DateRangePicker`, its necessary CSS, and register its module, as follows:

```ts
import { defineComponents, IgcDateRangePickerComponent } from 'igniteui-webcomponents';
import 'igniteui-webcomponents/themes/light/bootstrap.css';

defineComponents(IgcDateRangePickerComponent);
```

<!-- end: WebComponents -->

Now you can start with a basic configuration of the {Platform} `DateRangePicker`.

```html
<igc-date-range-picker anchor="hover-button">
  Congrats you've hovered the button!
</igc-date-range-picker>

<igc-button id="hover-button">Hover me</igc-button>
```

For a complete introduction to the {ProductName}, read the [*Getting Started*](../general-getting-started.md) topic.

## Usage

The `DateRangePicker` allows users to select a start and end date either by choosing a date range from a dropdown/calendar pop-up or by typing directly into the input fields - one for the start date and one for the end date.

> The picker offers two modes for displaying date values: single input and two inputs. In single input mode, the field is read-only, and the date range cannot be edited by typing. In two inputs mode, however, users can edit the start and end dates by typing in separate input fields.

Reviewer: ```suggestion
The picker offers two modes for displaying date values: single input and two inputs. In single input mode, the field is non-editable, and the date range cannot be edited by typing. In two inputs mode, however, users can edit the start and end dates by typing in separate input fields.
```
After clarifying the states with the UX team, apparently the better term for this is "non-editable"

### Display Date Range Picker

To instantiate a `DateRangePicker` in its default single input mode, use the following code:

```html
<igc-date-range-picker>
</igc-date-range-picker>
```

To switch the `DateRangePicker` to use two inputs, set the `UseTwoInputs` property to `true`.

```html
<igc-date-range-picker use-two-inputs="true">
</igc-date-range-picker>
```

### Value

In addition to being selected or typed by the user, the range value of the `DateRangePicker` can also be set using the `Value` property. It's important to note that the value must follow the format: **{ start: startDate, end: endDate }**, where `startDate` and `endDate` are `Date` objects representing the selected range.

```ts
let dateRange = document.querySelector('igc-date-range-picker') as IgcDateRangePickerComponent;
let startDate: Date = new Date(2025, 4, 6);
let endDate: Date = new Date(2025, 4, 8);
dateRange.value = { start: startDate, end: endDate }
```

> 

Reviewer: ```suggestion

In addition, the value can be set as attribute. In this case it should represent an object that can be parsed correctly as JSON, where the `start` and `end` fields should have date values in the ISO 8601 format:

```html
<igc-date-range-picker value='{"start":"2025-01-01","end":"2025-01-02"}'>
<igc-date-range-picker/>
```
```

### Read-only

You can also make the `DateRangePicker` read-only, which disables changing the range value through both typing and calendar selection. This is useful when the range is assigned via the value attribute and is intended to be display-only. To enable this behavior, simply set the `ReadOnly` property.

```html
<igc-date-range-picker use-two-inputs="true" readonly>
</igc-date-range-picker>
```

> 

Reviewer: [These details were elaborated](https://github.com/IgniteUI/igniteui-webcomponents/wiki/Date-Range-Picker#8-readonly-state) later in the process of design&implementation and can be added here.
Also we can mention the `nonEditable` property, which as opposed to `readOnly`, only makes the input non-editable by typing, while still allowing selection through the calendar & clearing trough the clear icon.

For example, to display the date in a Japanese format, you can set the locale property like this:
```html
<igc-date-range-picker locale="ja-JP">
</igc-date-range-picker>
```

If you want to manually define the date format, you can use the `InputFormat` property by passing a custom format string:

```html
<igc-date-range-picker input-format="dd/MM/yy">
</igc-date-range-picker>
```

The `DisplayFormat` property also accepts a custom format string, but it only applies when the input field is idle (i.e., not focused). When the field is focused, the format reverts to the default or to the one defined by `InputFormat`, if both properties are used together:

```html
<igc-date-range-picker input-format="dd/MM/yy" date-format="yy/MM/dd">
</igc-date-range-picker>
```

### Calendar Layout and Formatting

You can further customize the pop-up calendar using various properties:

* The `Orientation` property allows you to set whether the calendar should be displayed vertically or horizontally.

* The `VisibleMonths` property controls how many months are visible at a time, with a value of either 1 or 2.

* The `ShowWeekNumbers` property enables or disables the week number column in the calendar.

> 

Reviewer: Hm, this list is given here but is not exhaustive. Wouldn't it be better to add all calendar properties here as table, as in the current Properties section.
This would make the latter shorter, as it currently rather looks rather like the API pages.

```html
<igc-date-range-picker orientation="vertical" visible-months="1" show-week-numbers="true">
</igc-date-range-picker>
```

### Min & Max

You can also set the `Min` and `Max` properties to restrict user input by disabling calendar dates outside the defined range. These properties act as validators, so even if the user manually types a date outside the range, the `DateRangePicker` will become invalid.

```html
<igc-date-range-picker min="2025-05-06" max="2025-05-10">
</igc-date-range-picker>
```

> ### Custom Date range

Reviewer: ```suggestion
### Custom & Predefined Date ranges
```

> You can also add custom date range chips to the calendar pop-up for faster range selection using the `CustomRanges` property. For example, you can create a custom date range chip to quickly select the range for the previous 7 days, ending with the current date.

Reviewer: ```suggestion
You can also add custom date range chips to the calendar pop-up for faster range selection using the `CustomRanges` property. For example, you can create a custom date range chip to quickly select the range for the previous 7 days, ending with the current date. In addition, by setting the `UsePredefinedRanges` property, a set of predefined ranges chips will be displayed along with the custom ones.
```

```ts
const today: Date = new Date();

const previousSeven = new Date(
  today.getFullYear(),
  today.getMonth(),
  today.getDate() - 7 
);
const lastWeek: CustomDateRange[] = [
  {
    label: 'Previous 7 days',
    dateRange: {
      start: previousSeven,
      end: today,
    },
  }
]

const dateRange = document.querySelector('igc-date-range-picker') as IgcDateRangePickerComponent;

> dateRange.customRanges = lastWeek

Reviewer: ```suggestion
dateRange.customRanges = lastWeek;
dateRange.usePredefinedRanges = true;
```

```

Now, when you click the newly created `"Previous 7 days"` chip in the calendar pop-up, the range will automatically be selected, starting from 7 days ago and ending today.

`sample="/scheduling/date-range-picker/custom-ranges", height="500", alt="{Platform} Date Range Picker Custom Ranges Example"`

### Forms

The `DateRangePicker` component can also be used seamlessly with the HTML form element. The `Min`, `Max`, and `Required` properties act as form validators.

> 

Reviewer: Maybe add info about configuring the disabled dates, i.e.:
```ts
const disabledDates: DateRangeDescriptor[] = [
  {
    type: DateRangeType.Between,
    dateRange: [minDate, maxDate],
  },
];
```
or link this calendar topic [section](https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/scheduling/calendar#disabled-dates), where more details about the `DateRangeDescriptor` objects are given.

When the calendar is visible, a date range can be selected by choosing both a start and end date. Selecting a date will set both the start and end date, and once a second date is chosen, it will set the end date. If a range is already selected, clicking any other date on the calendar will start a new range selection.

### Display Date Range Picker

To instantiate a `DateRangePicker` in its default single input mode, use the following code:

```html
<igc-date-range-picker>
</igc-date-range-picker>
```

To switch the `DateRangePicker` to use two inputs, set the `UseTwoInputs` property to `true`.

```html
<igc-date-range-picker use-two-inputs="true">
</igc-date-range-picker>
```

### Value

In addition to being selected or typed by the user, the range value of the `DateRangePicker` can also be set using the `Value` property. It's important to note that the value must follow the format: **{ start: startDate, end: endDate }**, where `startDate` and `endDate` are `Date` objects representing the selected range.

```ts
let dateRange = document.querySelector('igc-date-range-picker') as IgcDateRangePickerComponent;

> let startDate: Date = new Date(2025, 4, 6);
> let endDate: Date = new Date(2025, 4, 8);

Reviewer: ```suggestion
let startDate = new Date(2025, 4, 6);
let endDate = new Date(2025, 4, 8);
```
minor code style tip

### Forms

The `DateRangePicker` component can also be used seamlessly with the HTML form element. The `Min`, `Max`, and `Required` properties act as form validators.

`sample="/scheduling/date-range-picker/form", height="500", alt="{Platform} Date Range Picker Form Example"`

## Additional configuration

### Properties

In addition to the properties we've already covered, the `DateRangePicker` component offers a variety of additional properties that allow you to further configure its behavior.

<!-- WebComponents -->

|Name|Type|Description|
|--|--|--|
| `Disabled` | boolean | Disables the component. |
| `NonEditable` | boolean |	Disables typing in the input field(s). |
| `Placeholder` | string | Placeholder text for the single input mode. |
| `PlaceholderStart` | string | Placeholder text for the start date input (two inputs mode). |
| `PlaceholderEnd` | string | Placeholder text for the end date input (two inputs mode). |
| `Outlined` | boolean | Determines whether the input part will have outline appearance in the [Material theme](../themes/overview.md). |
| `Prompt` | string | The prompt character used for unfilled parts of the input(s) mask. |

> | `ResourceStrings` | IgcCalendarResourceStrings & IgcDateRangePickerResourceStrings | Resource strings for localization of the date-range picker and the calendar. |

Reviewer: ```suggestion
| `ResourceStrings` | IgcDateRangePickerResourceStrings | Resource strings for localization of the date-range picker and the calendar. |
```
that was recently refactored and I have updated the spec - https://github.com/IgniteUI/igniteui-webcomponents/wiki/Date-Range-Picker#options
